### DISCUSSION TITLE
Allow to change the env var name `DATABASE_URL` via `diesel.toml`

### DISCUSSION THREAD

**Discussion by cbeck88** [diesel-rs/diesel#3850] [Ideas]
I have a project with several services using several databases via diesel.
Because of this, I often have to use the diesel cli with several different projects, and it needs to know the database url.

Per [docs](https://github.com/diesel-rs/diesel/blob/master/diesel_cli/README.md):

```
You can then run your new migration by running diesel migration run. Your DATABASE_URL must be set in order to run this command, and there are several ways that you can set it:

    Set it as an environment variable manually
    Set it as an environment variable using [dotenv](https://github.com/dotenv-rs/dotenv#examples)
    Pass it directly by adding the --database-url flag
```

The workflow I have created involves using option 2 here -- each project has its own `.env` and when I `cd` to that directory, the diesel-cli picks up the right `DATABASE_URL` this way. This is much less onerous than passing via `--database-url` flag or resetting the environment variable by hand every time I change contexts.

The problem I have now is that we have a need to change the name of this env `DATABASE_URL` in the actual server configs. But, AFAICT, there is no way to make `diesel-cli` read a different env variable, it always reads only `DATABASE_URL`.

In my ideal world, we could override the name of the env that diesel cli reads by setting an option in `diesel.toml`, which is project specific.

I could instead have the `.env` in each directory reserved for the diesel cli, and a different env file which the server actually reads But then I will have to keep these in sync if the dev environment changes, and I'm sure that it will trip up some future engineer who won't know about the `.env` file which is read only by diesel cli, because it will become very obscure.

---

The other option I thought of is along these lines, using the variable expansion features of `dotenvy` / `dotenvs`:

```
MY_SERVICE_DATABASE_URL=postgres://dev@localhost/my_service
# Setting DATABASE_URL is required for diesel cli tool, but is now ignored by the actual servers and migration binaries.
DATABASE_URL=${MY_SERVICE_DATABASE_URL}
```

This works but it's a bit messy compared to the ease of setting something in `diesel.toml`.

---

How do you deal with this problem? Do you have a better suggestion?

**Comment by weiznich** [comment]
I think the most obvious solution here might be to just use the command line flag. In that way the used database url is not hidden for others.
Otherwise it might also be possible to just reassign the database url in your `.env` file like:

```
DATABASE_URL=$OTHER
```

(That works with `OTHER=your_db_url diesel setup)
